### ISSUE TITLE
glyph_width(code_point) throws exception if @by_codepoint doesn't contain a matching glyph

### ISSUE THREAD

**Issue by scytherswings** [yob/pdf-reader#162]
Here's the method that this bubbles up to:

```ruby
      def glyph_width(code_point)
        return 0 if code_point.nil? || code_point < 0

        m = @metrics.char_metrics_by_code[code_point]
        if m.nil?
          names = @font.encoding.int_to_name(code_point)
          m = names.map { |name|
            @metrics.char_metrics[name.to_s]
          }.compact.first
        end

        if m
          m[:wx]
        elsif @font.widths[code_point - 1]
          @font.widths[code_point - 1]
        elsif control_character?(code_point)
          0
        else
          0
        end
      end
```

I have a pdf that causes `unicode_to_name(codepoint)` to return `nil` which eventually results in `names` not being set and `names.map` will throw a `method not found` exception.

Here's the section: 
```ruby
    def unicode_to_name(codepoint)
      @by_codepoint[codepoint.to_i]
    end
```

My specific codepoint that caused this was `157` which will return `nil`

Here's a gist of my `@by_codepoint` hash that is missing this element:
https://gist.github.com/scytherswings/c4c8bf08bcc58b3979e9

**Comment by scytherswings** [comment]
If `glyph_width(code_point)` gets a nil check on that return value I think it should solve the problem:

```ruby
def glyph_width(code_point)
        return 0 if code_point.nil? || code_point < 0

        m = @metrics.char_metrics_by_code[code_point]
        if m.nil?
          names = @font.encoding.int_to_name(code_point)

          if names.nil?
            return 0
          end

          m = names.map { |name|
            @metrics.char_metrics[name.to_s]
          }.compact.first
        end

        if m
          m[:wx]
        elsif @font.widths[code_point - 1]
          @font.widths[code_point - 1]
        elsif control_character?(code_point)
          0
        else
          0
        end
      end
```

**Comment by yob** [comment]
Thanks for this, I've merged your fix from #163 with a small modification.
